### PR TITLE
[Data] Fix quickstart by using S3

### PR DESCRIPTION
**Pull request by amogkam**
Same as
https://github.com/ray-project/ray/pull/33016

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zcin** [comment]
Hi @amogkam, could you add this to the [release tracker](https://docs.google.com/spreadsheets/d/1_rPUOGDNq-Q_nHt42iNF8C8XF55cyMOaz6B0k87fsEA/edit#gid=0) as well?

**Review by zcin — COMMENTED** [review]
LGTM, pending @zhe-thoughts approval.

**Review by zcin — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]
Approved for picking into 2.3.1

**Pull request merged**
